docs/blog/posts/2024-03-29-standby-instance.md

```diff
@@ -0,0 +1,69 @@
+---
+draft: false
+date: 2024-03-29
+search:
+  exclude: true
+authors:
+  - gemini_1_5_pro
+  - fkorotkov
+categories:
+  - announcement
+---
+
+## Persistent Workers Get a Boost with Standby Instances
+
+We're excited to announce a new feature for Cirrus CI persistent workers: **Standby Instances**. This functionality significantly improves scheduling efficiency when dealing with many persistent workers, leading to faster task execution and a smoother CI/CD experience.
+
+### Background: Persistent Workers and Scheduling Challenges
+
+Cirrus CI's persistent workers offer a powerful way to run tasks on dedicated infrastructure, including bare metal and virtual machines. This is ideal for scenarios where you need more control over the environment or want to leverage specific hardware configurations.
+
+However, with many persistent workers, scheduling tasks efficiently can be challenging. Previously, when a task was assigned to a worker, **the persistent worker itself would**:
+
+1. **Pull an image:** This could take several minutes if the image is missing or outdated.
+2. **Boot the VM:** This added another 10-20 seconds to the overall task execution time.
+
+This delay could be frustrating, especially when dealing with frequent builds and deployments.
+
+### Introducing Standby Instances: Faster Scheduling, Smoother CI/CD
+
+Standby instances address this challenge by ensuring tasks are assigned to workers that are already prepared. Here's how it works:
```

Comment:
>Standby instances address this challenge by ensuring tasks are assigned to workers that are already prepared.

This doesn't seem to be true because the upstream doesn't know anything about the state of the Standby Instances on the Persistent Worker.

Reply:
Yes and no. It is ensured because a worker will only poll a task once it's "warm".